### PR TITLE
update pe js consumer examples

### PR DESCRIPTION
**Pull request by sksadjad**
**update pe js consumer examples**

- updated existing code with the new pejs interface
- added two simple scenarios to project:
  1. one: we have a presentation definition and three credentials. only one has the information stated in presentationDefinition
  2. two: we have a presentation definition and six credentials. two have the information stated in presentationDefinition

### CODE REVIEW

**Review by hrehman-sphereon — CHANGES_REQUESTED** [review]
We should have a strict flag true for the project now.

File: scripts/consumer-scenario-2.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  const evaluateResult: EvaluationResults = holderPE.evaluate(presentationDefinition, {
+    "@context": [],
+    type: [],
+    verifiableCredential: vcs,
+    holder: undefined
+  });
+  const selectFromResult: SelectResults = holderPE.selectFrom(presentationDefinition, vcs, [wallet1.holder])
+  const presentationSubmission: PresentationSubmission = holderPE.submissionFrom(presentationDefinition, selectFromResult.verifiableCredentials);
+  console.log("evaluateResult: ", JSON.stringify(evaluateResult, null, 2));
+  console.log("selectFromResult: ", JSON.stringify(selectFromResult, null, 2));
```
Comment (hrehman-sphereon) [inline scripts/consumer-scenario-2.ts:60]: The result seem to have

 ```
"matches": [
    {
      "rule": "all",
      "matches": [
        "$[2]",
        "$[5]"
      ]
    }
  ],

```

This does not seem correct.

**Review by hrehman-sphereon — CHANGES_REQUESTED** [review]
The response is creating a doubt on the implementation.
Referring to comment
"$[5]"

File: scripts/consumer-scenario-2.ts (hunk 1 above)
Comment (sksadjad) [reply scripts/consumer-scenario-2.ts:60]: that is the correct response. we have two valid credential for one thing
it's up to user to select and send one of them

**Comment by sksadjad** [comment]
> We should have a strict flag true for the project now.

added

File: scripts/consumer-scenario-2.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const presentationDefinition = getPresentationDefinition();
+  let holderPE: PEJS = new PEJS();
+  const vcs = [...wallet1.verifiableCredentials, ...wallet2.verifiableCredentials];
+  const evaluateResult: EvaluationResults = holderPE.evaluate(presentationDefinition, {
+    "@context": [],
+    type: [],
+    verifiableCredential: vcs,
+    holder: undefined as unknown as string
+  });
+  console.log("evaluateResult: ", JSON.stringify(evaluateResult, null, 2));
```
Comment (zoemaas) [inline scripts/consumer-scenario-2.ts:59]: I think that having too many things logged to the console makes it hard to keep track of what really matters. But it's just my opinion.

**Review by zoemaas — APPROVED** [review]
It looks good to me.

File: scripts/consumer-scenario-2.ts (hunk 1 above)
Comment (hrehman-sphereon) [reply scripts/consumer-scenario-2.ts:60]: "verifiableCredentials" has only 2 objects these indeces are incorrect. The correct values can only be '0' & '1'
Results file is attached to the ticket.

selectFromResult:  {
  "errors": [
    ...
  ],
  "matches": [
    {
      "rule": "all",
      "matches": [
        "$[2]",
        "$[5]"
      ]
    }
  ],
  "verifiableCredentials": [
    {
      ...
      "credentialSubject": {
        "id": "did:key:z6MkjRagNiMu91DduvCvgEsqLZDVzrJzFrwahc4tXLt9DoHd",
        "degree": {
          "type": "BachelorDegree",
          "name": "Bachelor of Science and Arts"
        }
      },
      ...
    },
    {
      ...
      "credentialSubject": {
        "id": "did:key:z6MkjRagNiMu91DduvCvgEsqLZDVzrJzFrwahc4tXLt9DoHd",
        "degree": {
          "type": "BachelorDegree",
          "name": "bachelor of applied science"
        }
      },
      ...
    }
  ],
  "warnings": []
}

File: scripts/consumer-scenario-2.ts (hunk 2 above)
Comment (hrehman-sphereon) [reply scripts/consumer-scenario-2.ts:59]: This is not production code. But rather to help developers. I would suggest to keep the details.

File: scripts/consumer-scenario-2.ts (hunk 1 above)
Comment (zoemaas) [reply scripts/consumer-scenario-2.ts:60]: Handled in [ESSIFI-158](https://sphereon.atlassian.net/browse/ESSIFI-158)

Comment (nklomp) [reply scripts/consumer-scenario-2.ts:60]: @hrehman-sphereon Do you confirm? If so please resolve

File: scripts/consumer-scenario-2.ts (hunk 2 above)
Comment (nklomp) [reply scripts/consumer-scenario-2.ts:59]: I agree.

**Review by sksadjad — COMMENTED** [review]
looks good to me

**Review by hrehman-sphereon — APPROVED** [review]

**Review by hrehman-sphereon — APPROVED** [review]

**Pull request merged**